packages/runtime/src/utils/deterministic-uuid.ts

```diff
@@ -0,0 +1,32 @@
+import crypto from 'crypto'
+import { canonicalize } from './canonicalize'
+
+/**
+ * Generates a deterministic UUID using SHA-256.
+ * @param value - The value to generate the UUID from.
+ * @param secretNamespace - A secret key used as the HMAC namespace (optional).
+ * @returns A deterministic UUID string.
+ */
+export function deterministicUUID(value: any): string {
```

Comment:
Could this be done with namespaced uuid V5?

Reply:
uuidv5 is using SHA-1. Because we need to make this unique per site, we need to also seed the siteId. But we don't have access to the siteId when the returned snapshot is 404. So I'm using x-site-api-key here.

Because I'm using x-site-api-key, I'm using the more secure hash instead of SHA-1. Do you think it's overkill and SHA-1 is secure enough?

Reply:
Ah, good thought. Personally, I'd be more concerned with keeping our codebase simple and robust over securing against the likelihood that someone uses a hash collision attack to crunch API key — that's a pretty computationally/time costly attack vector. That being said, this is a pretty well encapsulated utility function, so I'm not concerned either way.

Reply:
I totally agree with that. I just don't want to accidentally expose a security leak because we hash the API key. Let's keep this for now. Let me know if there's a more concise yet secure way to hash this.

Reply:
Oh. We could also not use the whole API key. For example, we could use the first half so we don't have to worry about the API key leaking. I'll update that.

Reply:
Updated to use uuid v5